samples/uart_echo.cpp

```diff
@@ -0,0 +1,86 @@
+/*
```

Comment:
It can be although this feature branch really enables the serialEvent callback stuff which this sample illustrates so it feels like it belongs to me. I don't have a strong opinion on it though.